sdk/include/opentelemetry/sdk/metrics/state/sync_metric_storage.h

```diff
@@ -101,8 +108,112 @@ class SyncMetricStorage : public MetricStorage, public WritableMetricStorage
                opentelemetry::common::SystemTimestamp collection_ts,
                nostd::function_ref<bool(MetricData &)> callback) noexcept override
   {
-    MetricData data;
-    if (callback(data))
+    opentelemetry::common::SystemTimestamp last_collection_ts = sdk_start_ts;
```

Comment:
Could this be moved to a separated .cc file as this function is quite big now?

Reply:
Yes, have moved it to a separate .cc file now.